core/src/main/java/org/eclipse/hono/util/ResourceLimits.java

```diff
@@ -41,21 +42,46 @@
      * 
      * @param maxConnections The maximum number of connections to set.
      * @return A reference to this for fluent use.
+     * @throws IllegalArgumentException if the maximum connections is set to less than -1.
      */
     public ResourceLimits setMaxConnections(final int maxConnections) {
+        if (maxConnections < -1) {
+            throw new IllegalArgumentException("maximum Connections can be set to >= -1");
```

Comment:
Captions and wording seems a bit off here ... Maybe "Maximum connections property must be set to value >= -1"